scripts/globals/events/starlight_celebrations.lua

```diff
@@ -0,0 +1,943 @@
+        local snowToken = player:hasKeyItem(xi.keyItem.SNOW_THEMED_GIFT_TOKEN)
+        local bellToken = player:hasKeyItem(xi.keyItem.BELL_THEMED_GIFT_TOKEN)
+        local starToken = player:hasKeyItem(xi.keyItem.STAR_THEMED_GIFT_TOKEN)
+        local reward = 0
+        local reward_table = {}
+        local gifts_table =
+            {
+                SNOW_TOKEN =
+                {
+                    5552,   -- Black Pudding
+                    18863,  -- Dream Bell
+                    5542,   -- Gateau Aux Fraises
+                    5616,   -- Lebkuchen House
+                    5622,   -- Candy Cane
+                    5621,   -- Candy Ring
+                    177,    -- Snowman Miner
+                    115,    -- Bastokan Tree
+                },
+                BELL_TOKEN =
+                {
+                    5552,   -- Black Pudding
+                    5621,   -- Candy Ring
+                    18863,  -- Dream Bell
+                    15752,  -- Dream Boots
+                    15753,  -- Dream Boots +1
+                    5620,   -- Roast Turkey
+                    178,    -- Snowman Mage
+                    116,    -- Windurstian Tree
+                },
+                STAR_TOKEN =
+                {
+                    5550,   -- Buche au Chocolat
+                    5622,   -- Candy Cane
+                    5621,   -- Candy Ring
+                    18863,  -- Dream Bell
+                    14519,  -- Dream Robe
+                    14520,  -- Dream Robe +1
+                    5542,   -- Gateau Aux Fraises
+                    5620,   -- Roast Turkey
+                    176,    -- Snowman Knight
+                    86,     -- San d'Orian Tree
+                },
+                RARE_ITEMS =
+                {
+                    18863,  -- Dream Bell
+                    14519,  -- Dream Robe
+                    14520,  -- Dream Robe +1
+                    15752,  -- Dream Boots
+                    15753,  -- Dream Boots +1
+                    176,    -- Snowman Knight
+                    177,    -- Snowman Miner
+                    178,    -- Snowman Mage
+                    86,     -- San d'Orian Tree
+                    115,    -- Bastokan Tree
+                    116,    -- Windurstian Tree
+                }
+            }
+
+        if invAvailable ~= 0 then
+            if snowToken then
+                player:delKeyItem(xi.keyItem.SNOW_THEMED_GIFT_TOKEN)
+                reward_table = gifts_table.SNOW_TOKEN
+            elseif bellToken then
+                player:delKeyItem(xi.keyItem.BELL_THEMED_GIFT_TOKEN)
+                reward_table = gifts_table.BELL_TOKEN
+            elseif starToken then
+                player:delKeyItem(xi.keyItem.STAR_THEMED_GIFT_TOKEN)
+                reward_table = gifts_table.STAR_TOKEN
+            end
+        else
+            player:messageSpecial(id.text.DEFAULT_CANNOT_BE_OBTAINED)
+            return
+        end
```

Comment:
This is 100% a nit, feel free to hit resolved and ignore.
I see this pattern a lot in this codebase, imo it's easier to read with early returns:
```suggestion
        if invAvailable == 0 then
            player:messageSpecial(id.text.DEFAULT_CANNOT_BE_OBTAINED)
            return
       end 

        if snowToken then
            player:delKeyItem(xi.keyItem.SNOW_THEMED_GIFT_TOKEN)
            reward_table = gifts_table.SNOW_TOKEN
        elseif bellToken then
            player:delKeyItem(xi.keyItem.BELL_THEMED_GIFT_TOKEN)
            reward_table = gifts_table.BELL_TOKEN
        elseif starToken then
            player:delKeyItem(xi.keyItem.STAR_THEMED_GIFT_TOKEN)
            reward_table = gifts_table.STAR_TOKEN
        end
```

Reply:
Updated!